### PR TITLE
468 tests for customer invoice line update mutation

### PR DESCRIPTION
**Pull request by andreievg**
Closes #468 

* Added a couple of more helpers (since this test seems quite hard, and wanted to simplify just so slightly)
* Added a bit more mock data
* Removed some not needed types in API/service (and changed name of `LineDoesntReferenceStockLine`   toLineDoesNotReferenceStockLine
* Not using invoice query inside mutation tests (just assert/unwrap)
* Add item change test for update supplier invoice line (and fixed associated logic error)
* Fix total after tax calculation for invoice line (globally)
* Added exhaustive check of update supplier invoice line (and fixed associated logic errors)

Found a weird bug, didn't have time to investigate: https://github.com/openmsupply/remote-server/issues/482

### CODE REVIEW

File: tests/graphql/delete_supplier_invoice_line.rs [hunk 1]
```diff
@@ -170,29 +169,20 @@ mod graphql {
...
-            InvoiceLineBelongsToAnotherInvoice(delete::InvoiceLineBelongsToAnotherInvoice {
-                description: "Invoice line belongs to another invoice".to_string(),
-                invoice: convert_graphql_client_type(invoice.data.unwrap().invoice)
-            },)
-        );
+        let error_variant = assert_unwrap_error!(response);
+        let invoice_variant =
+            assert_unwrap_enum!(error_variant, InvoiceLineBelongsToAnotherInvoice).invoice;
+        let invoice = assert_unwrap_enum!(invoice_variant, delete::InvoiceResponse::InvoiceNode);
+        assert_eq!(invoice.id, draft_supplier_invoice.id);
```
Comment (clemens-msupply) [inline tests/graphql/delete_supplier_invoice_line.rs:177]: Read your comment about macros but find it still hard to follow what is compared, maybe better with type hints in the IDE...

A nested match might be ugly but easier to follow? Sorry bit in a rush just thinking out loud

**Review by clemens-msupply — APPROVED** [review]

File: src/service/invoice_line/customer_invoice_line/update/validate.rs [hunk 2]
```diff
@@ -92,19 +95,21 @@ fn check_batch_exists_option(
...
-        // stock_line_id not changed in input
-        BatchPair {
-            main_batch: previous_batch,
-            previous_batch_option: None,
+        },
+        _ => {
+            // stock_line_id not changed
+            BatchPair {
+                main_batch: previous_batch,
+                previous_batch_option: None,
```
Comment (wlthomson) [inline src/service/invoice_line/customer_invoice_line/update/validate.rs:111]: I think `previous_batch` is fine, the `BatchPair` definition makes it clear that this is an `Option`?

Comment (andreievg) [reply src/service/invoice_line/customer_invoice_line/update/validate.rs:111]: or you can add to https://github.com/openmsupply/remote-server/issues/484

**Comment by andreievg** [comment]
closing as it will be merged by downstream branch
